addons/account_tax_python/models/account_tax.py

```diff
@@ -47,11 +48,30 @@ def compute_all(self, price_unit, currency=None, quantity=1.0, product=None, par
         for tax in self.filtered(lambda r: r.amount_type == 'code'):
             localdict = self._context.get('tax_computation_context', {})
             localdict.update({'price_unit': price_unit, 'quantity': quantity, 'product': product, 'partner': partner, 'company': company})
-            safe_eval(tax.python_applicable, localdict, mode="exec", nocopy=True)
+            tax.evaluate_expression_code(tax.python_applicable, localdict)
             if localdict.get('result', False):
                 taxes += tax
         return super(AccountTaxPython, taxes).compute_all(price_unit, currency, quantity, product, partner, is_refund=is_refund, handle_price_include=handle_price_include)
 
+    @api.constrains('python_compute', 'python_applicable')
+    def _check_python_compute_and_applicable(self):
+        product = self.env['product.product'].search([], limit=1)
+        for tax in self:
+            if tax.python_compute:
+                localdict = {'base_amount': 0, 'price_unit': 0, 'quantity': 1, 'product': product.id, 'partner': self.env.user.partner_id, 'company': self.env.company}
+                tax.evaluate_expression_code(tax.python_compute, localdict)
+            if tax.python_applicable:
+                localdict = {'price_unit': 0, 'quantity': 1, 'product': product.id, 'partner': self.env.user.partner_id, 'company': self.env.company}
+                tax.evaluate_expression_code(tax.python_applicable, localdict)
+
+    def evaluate_expression_code(self, code, localdict):
+        '''Checks the validity of python_compute and python_applicable'''
+        self.ensure_one()
+        try:
+            safe_eval(code, localdict, mode="exec", nocopy=True)
+        except Exception as e:
+            raise UserError(_("You entered invalid Python code in '%s'\n\n Error code: '%s'") % (self.name, str(e)))
```

Comment:
I'm not sure about this error message, because it might be raised after entering values for account move lines, for example. You should either change the message or don't use these emthods with try-block in  `_compute_amount`/`compute_all`